orchestration/src/test/java/com/sap/ai/sdk/orchestration/OrchestrationUnitTest.java

```diff
@@ -129,34 +137,68 @@ void testCompletion() {
                     .withHeader("Content-Type", "application/json")));
-    final var response = client.chatCompletion(prompt, config);
-    final var result = response.getOriginalResponse();
-    var llmChoice =
-        ((LLMModuleResultSynchronous) result.getOrchestrationResult()).getChoices().get(0);
-
-    final var groundingData =
-        (Map<String, Object>) result.getModuleResults().getGrounding().getData();
-    assertThat(groundingData.get("grounding_query")).isEqualTo("grounding call");
-    assertThat(groundingData.get("grounding_result").toString())
-        .startsWith("Joule is the AI copilot that truly understands your business.");
-    assertThat(result.getModuleResults().getGrounding().getMessage()).isEqualTo("grounding result");
-    assertThat(((ChatMessage) result.getModuleResults().getTemplating().get(0)).getContent())
-        .startsWith(
-            "What does Joule do? Use the following information as additional context: Joule is the AI copilot that truly understands your business.");
-    assertThat(llmChoice.getMessage().getContent())
-        .startsWith(
-            "Joule is an AI copilot that revolutionizes how users interact with their SAP business systems.");
-    assertThat(llmChoice.getFinishReason()).isEqualTo("stop");
-    assertThat(llmChoice.getMessage().getContent())
-        .startsWith(
-            "Joule is an AI copilot that revolutionizes how users interact with their SAP business systems.");
+
+    final var documentMetadata =
+        SearchDocumentKeyValueListPair.create()
+            .key("document metadata")
+            .value("2")
+            .selectMode(List.of(SearchSelectOptionEnum.IGNORE_IF_KEY_ABSENT));
+    final var databaseFilter =
+        DocumentGroundingFilter.create()
+            .id("arbitrary-user-defined-id")
+            .dataRepositoryType(DataRepositoryType.VECTOR)
+            .searchConfig(GroundingFilterSearchConfiguration.create().maxChunkCount(3))
+            .documentMetadata(List.of(documentMetadata))
+            .chunkMetadata(List.of(KeyValueListPair.create().key("chunk metadata").value("1")));
+    final var groundingConfigConfig =
+        GroundingModuleConfigConfig.create()
+            .inputParams(List.of("query"))
+            .outputParam("results")
+            .addFiltersItem(databaseFilter);
+    final var groundingConfig =
+        GroundingModuleConfig.create()
+            .type(GroundingModuleConfig.TypeEnum.DOCUMENT_GROUNDING_SERVICE)
+            .config(groundingConfigConfig);
+    final var configWithGrounding = config.withGroundingConfig(groundingConfig);
+
+    final Map<String, String> inputParams =
+        Map.of("query", "String used for similarity search in database");
+    final var prompt =
+        new OrchestrationPrompt(
+            inputParams,
+            Message.system("Context message with embedded grounding results. {{?results}}"));
+
+    final var response = client.chatCompletion(prompt, configWithGrounding);
```

Comment:
(Question)

This is supposed to be the expected convenience API usage, right?
Looks tough.

Reply:
no, this is generated code only.